specs/time_series_api.adoc

```diff
@@ -0,0 +1,285 @@
+                  interactions with the time-series db.
+            ***** Since the manager is a singleton, there is only one instance
+                  of it throughout the life of Tendrl/performance-monitoring
+                  application which means all this house-keeping is done only
+                  once during the lifecycle of the application.
+* The following apis will be exposed by Tendrl/performance-monitoring:
+  ** An api to list the monitored resources of a particular entity
+  ** An api to get statistics of any monitored resource from list in above api
+     for a particular entity.
+* The Tendrl/tendrl-api proxies to these apis and remains agnostic of the
+  time-series db specifics.
+
+=== Alternatives
+
+None
+
+=== Data model impact:
+
+Data in time-series db is stored as key-value pairs(utilization to timestamp)
+in its specific collection/table named by the fully qualified resource name
+and this is very specific to the way collectd's plugins send metrics to time
+series db and hence not much control can be exercised here unless we use
+custom collectd plugins rather than the collectd provided ones
+
+=== Impacted Modules:
+
+==== Tendrl API impact:
+
+The tendrl-api interfaces to the apis mentioned above under the section
+"Proposed change" and detailed below under "Notifications/Monitoring impact".
+
+==== Notifications/Monitoring impact:
+
+The api list to be exposed will grow with time. Currently, the following apis
+can be considered:
+
+To explore the resources that are monitored, we can use the api described
+below.
+
+Sample Request
+
+----------
+GET /monitoring/nodes/<node_name>/monitored_types
+----------
+
+Sample Response
+
+----------
+Status: 200 OK
+
+[{
+  type: string,
+  description: the resource name,
+}]
+
+ex:
+['resource_1', 'resource_2', ...]
+----------
+
+The above response which is an array of resource name strings gives different
+statistics that tendrl's monitoring stack maintains in its time series db.
+
+
+Sample Request
+
+----------
+GET /monitoring/nodes/<node_name>/<resource_name>/stats
+----------
+
+Sample Response
+
+----------
+Status: 200 OK
+[
+  {
+    "target": {
+      type: string,
+      description: full qualified name of the resource including the
+        node name, a default prefix which gives a namespace to the stats in
+        graphite and name of the resource.
+    },
+    "datapoints": [
+      [{
+          type: float,
+          description: Instant value of target
+        }, {
+          type: long,
+          description: Timestamp at which the value was observed
+      }],....
+    ]
+  }
+]
+
+ex: [
+  {
+    "target": "{graphite_prefix.host_name.metric_name}",
+    "datapoints": [
+      [5.50091681946991, 1478495700],....
+    ]
+  }
+]
+----------
+
+The above response gives the statistics for last 24 hours corresponding to the
+'resource_name' passed to it. If a time range is passed as optional arguements
```

Comment:
And speaking of which, what's the way to change the duration for which the data would be stored in the db? Can this be done on the fly while the system is still pushing updates to the time series db?